components/Post.php

```diff
@@ -101,6 +101,16 @@ protected function loadPost()
             return $this->controller->run('404');
         }
 
+        if ($post->categories->count() && $this->param('category')) {
+            if ($post->categories->pluck('slug')->search($this->param('category')) === false) {
+                $canonicalUrl = $this->controller->pageUrl($this->page->baseFilename, [
+                    'category' => $post->categories[0]->slug,
+                    'slug' => $post->slug
```

Comment:
```suggestion
                    $this->paramName('slug') => $post->slug,
```

Reply:
Maybe `'category'` should also be `$this->paramName('category')`, in that case?

Reply:
@pavsid unfortunately it doesn't work that way. It's looking at the defined properties to figure out what the value is and the category slug isn't a property. If we made the category slug a property then it could work.